### PR TITLE
Fix 3

### PR DESCRIPTION
**Pull request by navytux**
A couple of fixes found while running fuzzer again.

### CODE REVIEW

File: fuzz.go [hunk 1]
```diff
@@ -33,7 +33,10 @@ func Fuzz(data []byte) int {
 		if err != nil {
 			// must succeed, as obj was obtained via successful decode
 			// the only exception is that we cannot encode non-string Ref at proto=0
```
Comment (kisielk) [inline fuzz.go]: This comment is now out of date

Comment (navytux) [reply fuzz.go]: True, thanks for catching. I will post corrected patch.

**Comment by navytux** [comment]
patch amended with the following interdiff:

```diff
diff --git a/fuzz.go b/fuzz.go
index cbb043e..aae806e 100644
--- a/fuzz.go
+++ b/fuzz.go
@@ -32,11 +32,14 @@ func Fuzz(data []byte) int {
                err = enc.Encode(obj)
                if err != nil {
                        // must succeed, as obj was obtained via successful decode
-                       // the only exception is that we cannot encode non-string Ref at proto=0
+                       // some  exceptions are accounted for first:
                        switch {
                        case proto == 0 && err == errP0PersIDStringLineOnly:
+                               // we cannot encode non-string Ref at proto=0
                                continue
+
                        case proto <= 3 && err == errP0123GlobalStringLineOnly:
+                               // we cannot encode Class (GLOBAL opcode) with \n at proto <= 4
                                continue
                        }
                        panic(fmt.Sprintf("protocol %d: encode error: %s", proto, err))
```

**Comment by kisielk** [comment]
Thanks!

**Comment by navytux** [comment]
Thanks, too!

**Pull request merged**